drivers/serial/uart_npcx.c

```diff
@@ -90,9 +90,41 @@ static void uart_npcx_clear_rx_fifo(const struct device *dev)
 	while (uart_npcx_rx_fifo_available(dev))
 		scratch = inst->URBUF;
 }
-#endif
 
-/* UART api functions */
+/* Forward declaration to be used in uart_npcx_poll_in(). */
+static int uart_npcx_fifo_read(const struct device *dev, uint8_t *rx_data,
```

Comment:
Can you re-organize (ie, move `uart_npcx_poll_in` and `uart_npcx_poll_out` so they are after `uart_npcx_fifo_read` and `uart_npcx_fifo_fill`) so we don't need the forward declarations?

Reply:
Sure thing

Reply:
OK, I took the section below that was also guarded by `CONFIG_UART_INTERRUPT_DRIVEN` and merged it with the one above. I can split this up into 2 commits if it makes it easier to review. Let me know.

Reply:
Yeah, 2 commits would be better, one that just moves the code and the second that makes the changes.

Reply:
Sure, I'll upload ASAP

Reply:
Done